src/components/map/MapUser.js

```diff
@@ -1,12 +1,232 @@
+  }
+
+  redirectToChat() {
+    let url = "/chat/" + this.state.user.id.toString();
+    this.props.history.push(url);
+  }
+
   render() {
+    let iconUrl = "";
+    if (this.state.user.status === "ONLINE") {
+      iconUrl = "/images/map/marker-online.png";
+    } else {
+      iconUrl = "/images/map/marker-offline.png";
+    }
+
+    // TODO: Fix map alignment (oversheets screen bottom atm.)
     return (
-      <h1>Show User ID {this.props.match.params.id} on the map</h1>
-    );
+      <div className="h-screen flex flex-col">
+        <div className="p-4 bg-gray-100">
+          <div className="flex">
+            <div className="flex-none mr-4">
+              <img src="https://upload.wikimedia.org/wikipedia/en/6/64/Cruella_de_Vil.png" className="h-24 w-24 rounded-full bg-gray-400"></img>
+            </div>
+            <div>
+              <div>
+                <h2 className="font-bold text-2xl">{this.state.user.name}</h2>
+              </div>
+              <div>
+                <StatusIndicator status={this.state.user.status} />
+                <span className="text-xs font-bold text-gray-500 ml-2">{this.state.user.distance}</span>
+              </div>
+              <div>
+                <p>{this.state.user.bio}</p>
+              </div>
+            </div>
+          </div>
+          <div>
+            <h2 className="font-bold text-lg mb-2 mt-4">DOGS</h2>
+            <div className="flex flex-wrap">
+              {this.state.user.dogs.map(dog => {
+                return (
+                  <div key={dog.id} className="w-1/2">
+                    <Dog name={dog.name} sex={dog.sex} breed={dog.breed} age={dog.age} imageUrl={dog.imageUrl}></Dog>
+                  </div>   
+                )
+              })}
+            </div>
+          </div>
+          <div>
+            <Button onClick={this.redirectToChat}>💬 Chat</Button>
+            <Button onClick={this.redirectToProfile}>View Full Profile</Button>
+          </div>
+        </div>
+      
+        <div className="flex-grow">
+          <GoogleMap
+              google={this.props.google}
+              zoom={18}
+              containerStyle={mapStyles}
+              center={this.state.user.location}
+              fullscreenControl={false}
+              mapTypeControl={false}
+          >
+
+            <Marker
+              title={this.state.user.name}
+              name={this.state.user.name}
+              id={this.state.user.id}
+              key={this.state.user.id}
+              position={this.state.user.location}
+              onClick={this.onMarkerClick}
+              icon={iconUrl} />
+          </GoogleMap>
+        </div>
+
+        <div className="absolute inset-x-0 bottom-0">
+          <TabBar active="map" />
+        </div>
+      </div>
+    )
   }
 }
 
-export default withRouter(MapUser);
\ No newline at end of file
+export default withRouter(
+  GoogleApiWrapper({
+    apiKey: 'AIzaSyA0LrnCgNKy53NOcrnzzUkHJxeD5eyeWT4'
```

Comment:
AFAIK there is no way of truly hiding this key since it will be needed by the client anyway. There should be an option to restrict the valid referrer domains for the key in the google console once we deploy the app, so that it cannot be used from another domain.